sgkit/stats/ibs.py

```diff
@@ -0,0 +1,260 @@
+from typing import Hashable
+
+import dask.array as da
+from numba import guvectorize
+from xarray import Dataset
+
+from sgkit import variables
+from sgkit.stats.aggregation import infer_call_allele_frequencies
+from sgkit.typing import ArrayLike
+from sgkit.utils import (
+    conditional_merge_datasets,
+    create_dataset,
+    define_variable_if_absent,
+)
+
+
+@guvectorize(  # type: ignore
+    [
+        "void(float32[:,:], float32[:,:], float32[:,:])",
+        "void(float64[:,:], float64[:,:], float64[:,:])",
+    ],
+    "(x,a),(y,a)->(x,y)",
+    nopython=True,
+    cache=True,
+)
+def identity_by_state(
```

Comment:
Not something we need to follow up on here, but I bet this would work well on the GPU backend.